packages/@aws-cdk/aws-appsync/lib/schema-base.ts

```diff
@@ -164,7 +174,36 @@ export class Directive {
    * Add the @aws_iam directive
    */
   public static iam(): Directive {
-    return new Directive('@aws_iam');
+    return new Directive('@aws_iam', AuthorizationType.IAM);
+  }
+
+  /**
+   * Add the @aws_oidc directive
+   */
+  public static oidc(): Directive {
+    return new Directive('@aws_oidc', AuthorizationType.OIDC);
+  }
+
+  /**
+   * Add the @aws_api_key directive
+   */
+  public static apiKey(): Directive {
+    return new Directive('@aws_api_key', AuthorizationType.API_KEY);
+  }
+
+  /**
+   * Add the @aws_auth or @aws_cognito_user_pools directive
+   *
+   * @param groups the groups to allow access to
+   */
+  public static cognito(...groups: string[]): Directive {
+    if (groups.length === 0) {
+      throw new Error('Groups parameter must not be empty.');
```

Comment:
modify this error message so it provides a little more information and is prescriptive in the context of a synth error.

i.e. Cognito authorization requires at least one Cognito group to be supplied. Received: 0

or whatever you choose